### PR TITLE
Remove outdated ansible managed header and use {{ ansible_managed | comment }}

### PR DESCRIPTION
**Pull request by spetrosi**
I am testing with ansible_managed set as such in ansible.cfg:
```
ansible_managed = WARNING: This file is managed by ANSIBLE.
  template: {file}
  modified on: %Y-%m-%d %H:%M:%S
  by user: {uid}
  from ansiblemaster: {host}
```
This results in the following at `/etc/postfix/main.cf`:
```
# BEGIN ANSIBLE MANAGED BLOCK
#
# WARNING: This file is managed by ANSIBLE.
# template: /home/spetrosi/Documents/work/linux-system-roles/postfix/tests/roles/linux-system-roles.postfix/templates/get_ansible_managed.j2
# modified on: 2022-02-17 16:15:39
# by user: spetrosi
# from ansiblemaster: fedora
#
# END ANSIBLE MANAGED BLOCK
```
It is dynamic because it uses date, which means that it is never idempotent. This is, however, how the upstream reporter of the ansible_managed issue in https://github.com/linux-system-roles/timesync/issues/123 used it.
I came up with `blockinfile` because `lineinfile` is not able to replace the existing multi-line ansible_managed block, it just pastes another one above.

### CODE REVIEW

**Comment by spetrosi** [comment]
@richm I fixed the redundant `trim`. That was a very good catch, thank you.

File: tasks/main.yml [hunk 1]
```diff
@@ -49,14 +49,20 @@
...
-        state: present
+        regexp: "# This file is managed by [aA]nsible"
+        state: absent
+
+    - name: Ensure ansible_managed header in configuration file
+      vars:
+        __postfix_ansible_managed: "{{
+          lookup('template', 'get_ansible_managed.j2') | trim }}"
+      blockinfile:
+        dest: /etc/postfix/main.cf
```
[the remark this answers is not in the document]
Comment (nhosoi) [reply tasks/main.yml:65]: I thought using the newer name `path` would be better as we don't have to support Ansible 2.3-. Plus `dest` could become deprecated some day in the future. But I guess you think using the older name covers wider varieties. :) Well, as long as the `aliaes` are ensured to exist, it does not matter. So, I guess it's ok to leave it as it is. Thanks.

[the remark this answers is not in the document]
Comment (spetrosi) [reply tasks/main.yml:65]: Sorry Noriko I misread everything, yes I will update to path as it's newer. I thought I was using path already, sorry.

**Review by nhosoi — APPROVED** [review]
lgtm
Thanks, @spetrosi !

**Comment by spetrosi** [comment]
@nhosoi I guess we do not need tests for ansible_managed, am I right?
I looked at the previous ansible_managed PRs and did not find ones similar to what I did here. Other PRs either put ansible_managed in templates, or via Ansible modules. Please correct me if I am wrong.

**Comment by spetrosi** [comment]
I noticed a bug with this solution, if I put time into the `ansible_managed` definition in ansible.cfg, the role puts the current time stamp. So the header in the file is as is but does not update with the further role runs. I'll try to think what to do with this, please suggest your ideas if any too :)
```
[root@ci-vm-10-0-138-91 ~]# head /etc/postfix/main.cf
# BEGIN ANSIBLE MANAGED BLOCK
#
# WARNING: This file is managed by ANSIBLE.
# template: get_ansible_managed.j2
# modified on: 2022-02-21 09:03:49
# by user: spetrosi
# from ansiblemaster: fedora
#
# END ANSIBLE MANAGED BLOCK
# Global Postfix configuration file. This file lists only a subset
```

**Comment by nhosoi** [comment]
> I noticed a bug with this solution, if I put time into the `ansible_managed` definition in ansible.cfg, the role puts the current time stamp. So the header in the file is as is but does not update with the further role runs. I'll try to think what to do with this, please suggest your ideas if any too :)

Ahhh, it's so tricky, isn't it? The issue is shared among multiple roles now... If the timestamp format is fixed like `%Y-%m-%d %H:%M:%S`, we could just compare the config files after eliminating the timestamp part and learn if the rest of the contents are identical or not. But the timestamp format is quite flexible. I'm not sure if there's a perfect way to delete just the timestamp from the header... :(

**Comment by richm** [comment]
This is why it is not a good practice to put a timestamp in the ansible_managed header - https://github.com/redhat-cop/automation-good-practices/blob/main/roles/README.adoc#generating-files-from-templates
```
When commenting, don’t include anything like "Last modified: {{ date }}". This would change the file at every application of the role, even if it doesn’t need to be changed for other reasons, and thus break proper change reporting.
```

**Comment by richm** [comment]
So if the user wants ansible_managed to include a timestamp, then they will have to know that they are breaking role idempotency and check_mode - we should not do anything to prevent this in the roles, we should just suggest that this is not recommended practice because it breaks idempotency and check_mode

**Comment by nhosoi** [comment]
> This is why it is not a good practice to put a timestamp in the ansible_managed header - https://github.com/redhat-cop/automation-good-practices/blob/main/roles/README.adoc#generating-files-from-templates
> 
> ```
> When commenting, don’t include anything like "Last modified: {{ date }}". This would change the file at every application of the role, even if it doesn’t need to be changed for other reasons, and thus break proper change reporting.
> ```

So, can we refer the good practice somewhere in the rhel-/linux-system-roles doc and leave the ansible_managed code as it is?

**Review by richm — APPROVED** [review]
lgtm

**Comment by spetrosi** [comment]
I must have been unclear, I am not talking about idempotency, it's fine that idempotency doesn't work with timestamps in ansible_managed, I do not think that we should mention it as it's a common Ansible knowledge.

The issue is that time is displayed incorrectly. I am running the following task against 1minutetip:
```
- debug:
    var: __postfix_ansible_managed
  vars:
    __postfix_ansible_managed: "{{
      lookup('template', 'get_ansible_managed.j2') }}"
```
I run it yesterday the first time, since then the output shows the same result with the same timestamp:
```
TASK [linux-system-roles.postfix : debug] ****************************************************************************************************************************************************
ok: [10.0.139.83] => {
    "__postfix_ansible_managed": "#\n# WARNING: This file is managed by ANSIBLE.\n# template: get_ansible_managed.j2\n# modified on: 2022-02-21 09:03:49\n# by user: spetrosi\n# from ansiblemaster: fedora\n#\n"
```
I expect the timestamp to change each time I am running the playbook, but it does not.

**Comment by spetrosi** [comment]
After I reboot my laptop, I see the same `2022-02-21 09:03:49` timestamp on the output... I think I can merge this PR now and deal with this weird issue after because the role works correctly the first time.

File: tasks/main.yml [hunk 2]
```diff
@@ -45,18 +45,26 @@
...
+        regexp: "# This file is managed by [aA]nsible"
+        state: absent
+
+    - name: Ensure ansible_managed header in configuration file
+      vars:
+        __postfix_ansible_managed: "{{
+          lookup('template', 'get_ansible_managed.j2') }}"
+      blockinfile:
+        path: /etc/postfix/main.cf
+        block: "{{ __postfix_ansible_managed }}"
```
Comment (nhosoi) [inline tasks/main.yml:66]: @spetrosi, a question came up as I was going to borrow this code for mssql. ;)
Does this `blockinfile` add `__postfix_ansible_managed` every time  when this `Ensure ansible_managed header in configuration file` task is executed?
1) when `/etc/postfix/main.cf` is bland new (without `# This file is managed by [aA]nsible`), it's good.
2) when `/etc/postfix/main.cf` already has the same header as `__postfix_ansible_managed` contains, `blockinfile` does nothing, I guess. Then, it's good.
3) when `/etc/postfix/main.cf` already has a different header from what `__postfix_ansible_managed` contains, can the existing header be replaced with the new one? It's not so clear to me...

The kernel_settings role captures the existing header and compares it with the new one. Based upon the result, it determines to add, replace or skip [0]. Do you think it is needed for `mssql`? How about here?
[0] - https://github.com/linux-system-roles/kernel_settings/blob/master/tasks/main.yml#L48-L102

Comment (richm) [reply tasks/main.yml:66]: try writing a simple test using `blockinfile` to see if it changes the header if different, and is idempotent - it is supposed to be (like all ansible modules) - also take a look at the source code for `blockinfile`

Comment (nhosoi) [reply tasks/main.yml:66]: Sorry, I should have... :p
Basically, `blockinfile` adds its own BEGIN and END and manage the strings in between. So, my concerns are all cleared. Thanks!
```
# BEGIN ANSIBLE MANAGED BLOCK
#
# New ansible managed
# Updated by nhosoi
#
# END ANSIBLE MANAGED BLOCK
[test]
abc: xyz
```

**Review by richm — APPROVED** [review]
lgtm

**Comment by spetrosi** [comment]
Did `s/__postfix_ansible_managed/__lsr_ansible_managed` for consistency with other lsr roles and rebased. Merging now, thank you for your reviews.

**Pull request merged**